### PR TITLE
Demo mode

### PR DESCRIPTION
**Pull request by peuter**
build source version with:

`./generate.py build --macro=CV_TESTMODE:resource/demo/media/demo_testmode_data.json`

### CODE REVIEW

File: utils/travis/deploy.sh [hunk 1]
```diff
@@ -92,9 +92,12 @@ ${CV} sitemap
 
 echo "generating test mode build"
 source temp-python/bin/activate
-./generate.py build --macro=CV_TESTMODE:resource/demo/media/metal-data.json
+./generate.py build --macro=CV_TESTMODE:resource/demo/media/demo_testmode_data.json
 rm -rf out/de/$VERSION/demo
 mv build out/de/$VERSION/demo
+
+# Copy demo-mode to default config
+cp out/de/$VERSION/demo/resource/demo/visu_config_demo_testmode.xnl out/de/$VERSION/demo/config/visu_config.xml
```
Comment (ChristianMayer) [inline utils/travis/deploy.sh]: `xnl`?

Comment (peuter) [reply utils/travis/deploy.sh]: Oops, typo!

File: source/resource/designs/metal/basic.css [hunk 2]
```diff
@@ -76,12 +76,16 @@ body br
   text-decoration:none;
 }
 
-.footer,
-.footer *
-{
-  color: #555;
+.footer, .footer {
```
Comment (ChristianMayer) [inline source/resource/designs/metal/basic.css]: `.footer, .footer *` !== `.footer, .footer`

`.footer, .footer` should be simplified to `.footer`

**Comment by ChristianMayer** [comment]
Does this PR (especially commit 4c46bac) close #785?

**Comment by peuter** [comment]
No, currently I have no fix for #785. Commit 4c46bac fixes definitly fixes an error, but I don't know which one. Stumbled open the code-line during searching for a fix for #785, but it did not help.

**Comment by ChristianMayer** [comment]
- [ ] Temperatures should be different between the rooms
- [ ] All rooms should exists (Büro/Kinder/Flur)
- [ ] Slider doesn't slide by mouse (Chrome), similar to the tablet issue
- [ ] pressed/unpressed button state looks too similar. The difference isn't easily visible
- [ ] Wohnen: Licht vs. Licht mit Icons - it would be better to show what's possible in a "show off demo" than to educate about subtle differences
- [ ] Refresh doesn't make sense in such a demo
- [ ] Make demo more interactive, i.e. let switches switch something somewhere else to give better feedback (e.g. ColorChooser -> r+g+b slider; main light slider -> measured brightness in Lux; ...)

Mode possibilities (requiring deeper code changes):
- [ ] Add time dimension (+ infinite loop) to testmode data json so that something is happening
- [ ] Add "callback functions" to testmode data which can react to sent data by sending something else back. This could allow an interactive, simulated dim or shade. Even complex behavior like room temperature could be simulated then

**Pull request merged**